### PR TITLE
Render conditionals for Cloud specific elements

### PR DESCRIPTION
**Pull request by Gaya**
We need to be able to hide certain UI elements when conditions apply.

## Description
In the case of cloud we need to be able to hide elements and disable whole pages altogether. This requires a conditional check, a higher order component that allows for easy hiding, and a function to filter any pages from registering in the app's menu and routes.

## Motivation and Context
To allow hiding / changing certain parts of the UI

## How Has This Been Tested?
Unit tested and tested using to hide specific elements in the UI and complete pages / routes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/util/conditional/isCloud.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// @flow strict
+
+import AppConfig from '../AppConfig';
+
+function isCloud(): boolean {
```
Comment (edmundoa) [inline graylog2-web-interface/src/util/conditional/isCloud.js:5]: I'm failing to see why we need this at the moment, could you please explain what the reasoning behind it is?

File: graylog2-web-interface/src/util/conditional/filterMenuItems.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// @flow strict
+
+import isCloud from './isCloud';
+
+type MenuItem = { path: string };
+
+function filterMenuItems(
+  menuItems: Array<MenuItem>,
+  toFilter: Array<string>,
```
Comment (edmundoa) [inline graylog2-web-interface/src/util/conditional/filterMenuItems.js:9]: Could we make it clearer that the array should contain the `path`s to filter? I think renaming the parameter to `pathsToFilter` would be enough.

**Review by edmundoa — COMMENTED** [review]
I added a couple comments but it looks good to me overall.

File: graylog2-web-interface/src/util/conditional/filterMenuItems.js (hunk 2 above)
Comment (Gaya) [reply graylog2-web-interface/src/util/conditional/filterMenuItems.js:9]: But wouldn't that imply that the array just contains paths, and not the whole menu items? I'd change it to `Array<string>` and just ask for the paths, makes it more reusable too. What do you think?

File: graylog2-web-interface/src/util/conditional/isCloud.js (hunk 1 above)
Comment (Gaya) [reply graylog2-web-interface/src/util/conditional/isCloud.js:5]: It was a fallback for when we didn't have the `AppConfig`. I'll refactor it out.

File: graylog2-web-interface/src/util/conditional/filterMenuItems.js (hunk 2 above)
Comment (edmundoa) [reply graylog2-web-interface/src/util/conditional/filterMenuItems.js:9]: I think making both arguments take the same data structure makes it clearer, so I like changing them to `Array<string>` and passing paths in this particular case.

Comment (Gaya) [reply graylog2-web-interface/src/util/conditional/filterMenuItems.js:9]: Will do that, thanks!

**Review by edmundoa — APPROVED** [review]
LGTM 👍

**Pull request merged**